### DISCUSSION TITLE
How to prevent overwriting?

### DISCUSSION THREAD

**Discussion by Ekrizdis367** [JanDeDobbeleer/oh-my-posh#3686] [Troubleshoot]
Hello!

Is it possible to prevent long text/commands from overlapping the right-aligned items, and instead breaking to a new line (wrapping) just before?

Or is the best way to avoid this to have it start the prompt on the line below?

**Comment by JanDeDobbeleer** [comment]
@Ekrizdis367 that depends on the shell you use. If they do not offer out of the box right prompt functionality (and handle it correctly), we print a right prompt manually which gets overridden. The additional logic to identify we're there and hide is shell side, and not something worthwhile to invest time in (it's not easy, and should be a shell's responsibility). If you're talking about a right aligned block (not `rprompt`), that behaviour is controlled by the [overflow](https://ohmyposh.dev/docs/configuration/block#overflow) property.
